app/src/main/java/com/nextcloud/talk/jobs/ContactAddressBookWorker.kt

```diff
@@ -143,9 +144,22 @@ class ContactAddressBookWorker(val context: Context, workerParameters: WorkerPar
                     }
 
                     override fun onNext(foundContacts: ContactsByNumberOverall) {
-                        val contactsWithAssociatedPhoneNumbers = foundContacts.ocs!!.map
-                        deleteLinkedAccounts(contactsWithAssociatedPhoneNumbers)
-                        createLinkedAccounts(contactsWithAssociatedPhoneNumbers)
+                        when (foundContacts.ocs?.meta?.statusCode) {
+                            HTTP_CODE_TOO_MANY_REQUESTS -> {
+                                Toast.makeText(
```

Comment:
fine for me to use a toast here as we have no view. 
just an edge case here which might never(?) be triggered..
otherwise we prefer snackbar